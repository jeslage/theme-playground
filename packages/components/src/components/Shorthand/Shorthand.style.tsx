import styled from 'styled-components';

const StyledShorthand = styled.div`
  display: flex;
  align-items: center;
  width: 100%;

  .shorthand__shorthand {
    display: flex;
    align-items: center;
  }

  input {
    border: 1px solid ${({ theme }) => theme.text};
    border-radius: 30px;
    outline: none;
    display: inline-block;
    max-width: 50px;
    padding: 5px 5px 4px;
    margin: 0 0 0 10px;
    font-size: 14px;
    font-family: inherit;
    text-align: center;
    color: inherit;
    background: none;
  }
`;

export default StyledShorthand;
