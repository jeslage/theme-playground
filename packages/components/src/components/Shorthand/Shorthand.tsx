import * as React from 'react';
import { HandleChange } from '../../definitions';

import StyledShorthand from './Shorthand.style';
import Label from '../Label/Label';

export interface Props {
  iconBefore?: JSX.Element;
  label?: string;
  name?: string;
  description?: string;
  onChange: (val: ShorthandObject) => void;
  value: ShorthandObject;
}

interface ShorthandObject {
  top: number | string;
  left: number | string;
  right: number | string;
  bottom: number | string;
}

const Shorthand: React.FC<Props> = ({
  label,
  description,
  value,
  onChange,
  iconBefore
}) => {
  const { top, left, right, bottom } = value;
  const updateValue = (key: string, val: number) => {
    const newValue = {
      ...value,
      [key]: val
    };

    onChange(newValue);
  };

  const handleChange = (event: HandleChange) => {
    const { value: eventValue, name } = event.target;

    if (eventValue !== '') {
      updateValue(name, parseFloat(eventValue));
    } else {
      updateValue(name, 0);
    }
  };

  return (
    <StyledShorthand>
      <Label iconBefore={iconBefore} label={label} description={description} />

      <div className="shorthand__shorthand">
        <input
          type="text"
          pattern="[0-9.]*"
          value={top}
          name="top"
          onChange={handleChange}
          onBlur={handleChange}
        />

        <input
          type="text"
          pattern="[0-9.]*"
          value={right}
          name="right"
          onChange={handleChange}
          onBlur={handleChange}
        />

        <input
          type="text"
          pattern="[0-9.]*"
          value={bottom}
          name="bottom"
          onChange={handleChange}
          onBlur={handleChange}
        />

        <input
          type="text"
          pattern="[0-9.]*"
          value={left}
          name="left"
          onChange={handleChange}
          onBlur={handleChange}
        />
      </div>
    </StyledShorthand>
  );
};

export default Shorthand;
