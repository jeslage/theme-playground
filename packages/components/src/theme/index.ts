export const darkTheme = {
  base: 'dark',
  backgroundDark: '#2f2f2f',
  backgroundLight: '#333',
  border: '#999999',
  text: '#FFFFFF'
};

export const lightTheme = {
  base: 'light',
  backgroundDark: '#f6f9fc',
  backgroundLight: '#fff',
  border: '#666666',
  text: '#333333'
};

export const getUiTheme = theme => {
  if (theme === undefined) return darkTheme;

  if (typeof theme === 'string') {
    if (theme === 'dark') {
      return darkTheme;
    } else if (theme === 'light') {
      return lightTheme;
    }

    throw Error(
      `Theme "${theme}" is not available. Please use "light" or "dark", or provide a custom theme object.`
    );
  }

  if (theme.base && theme.base === 'light') {
    return {
      ...lightTheme,
      ...theme
    };
  }

  return {
    ...darkTheme,
    ...theme
  };
};
