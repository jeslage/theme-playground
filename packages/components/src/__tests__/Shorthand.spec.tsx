import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';

import Shorthand from '../components/Shorthand/Shorthand';
import StyledShorthand from '../components/Shorthand/Shorthand.style';
import { darkTheme } from '../theme';

describe('Shorthand', () => {
  it('Shorthand should render correctly', () => {
    const component = shallow(
      <Shorthand
        onChange={value => console.log(value)}
        label="Label"
        description="Testbeschreibung"
        value={{ top: 20, left: 20, right: 20, bottom: 20 }}
        iconBefore={
          <svg height="24" viewBox="0 0 24 24" width="24">
            <rect height="2" rx="1" width="12" x="6" y="11" />
          </svg>
        }
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('StyledShorthand should render correctly', () => {
    const component = shallow(
      <StyledShorthand theme={{ colors: darkTheme }} />
    );

    expect(component).toMatchSnapshot();
  });
});
