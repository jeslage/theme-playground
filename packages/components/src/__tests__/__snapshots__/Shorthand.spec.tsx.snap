// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shorthand Shorthand should render correctly 1`] = `
<styled.div>
  <Label
    description="Testbeschreibung"
    iconBefore={
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <rect
          height="2"
          rx="1"
          width="12"
          x="6"
          y="11"
        />
      </svg>
    }
    label="Label"
  />
  <div
    className="shorthand__shorthand"
  >
    <input
      name="top"
      onBlur={[Function]}
      onChange={[Function]}
      pattern="[0-9.]*"
      type="text"
      value={20}
    />
    <input
      name="right"
      onBlur={[Function]}
      onChange={[Function]}
      pattern="[0-9.]*"
      type="text"
      value={20}
    />
    <input
      name="bottom"
      onBlur={[Function]}
      onChange={[Function]}
      pattern="[0-9.]*"
      type="text"
      value={20}
    />
    <input
      name="left"
      onBlur={[Function]}
      onChange={[Function]}
      pattern="[0-9.]*"
      type="text"
      value={20}
    />
  </div>
</styled.div>
`;

exports[`Shorthand StyledShorthand should render correctly 1`] = `
<div
  className="sc-AxjAm fTDmTT"
/>
`;
